### PR TITLE
Workflow assignment dialog

### PR DESCRIPTION
**Pull request by srallen**
This adds the dialog UI for workflow assignment/promotion as well as a bug fix for the Tutorial and removed a redundant API call. Also, updated the eslint file and changed it to `.eslintrc` so it would work with Sublime.

Still need to:
- Finalize copy for the dialog
- Finalize any styling for the dialog
- Do some additional testing when I can successfully get Project Preferences of another user and promote them (via python client)

This leaves out the potential opt out feature. I think that's gonna take more thought and time if we do it at all. FYI @trouille 

Staged at: https://workflow-assignment-dialog.pfe-preview.zooniverse.org/projects/srallen086/workflow-assignment-testing

### CODE REVIEW

**Comment by srallen** [comment]
Content and styling are done.

**Comment by srallen** [comment]
Rebased to include the workflow level buttons feature.

File: app/pages/project/classify.cjsx [hunk 1]
```diff
@@ -92,6 +94,16 @@ module.exports = React.createClass
   componentWillUnmount: () ->
     @context.geordi?.forget ['workflowID']
 
+  componentWillReceiveProps: (nextProps) ->
+    @shouldWorkflowAssignmentPrompt(nextProps.preferences)
+
+  shouldWorkflowAssignmentPrompt: (preferences) ->
+    # Only for Gravity Spy which is assigning workflows to logged in users
+    assignedWorkflowID = preferences?.settings?.workflow_id
+    if @props.project.experimental_tools.indexOf 'workflow assignment' > -1 and @props.user?
```
Comment (mcbouslog) [inline app/pages/project/classify.cjsx]: This `indexOf` doesn't have the parentheses around `'workflow assignment'`, but it seems to be working fine, so I'm not sure if the weird Coffeescript and indexOf compiling thing is actually happening. Maybe the `if assignedWorkflowID?` conditional below is stopping it from causing problems?

Comment (srallen) [inline app/pages/project/classify.cjsx]: Probably would be a good idea to include the parentheses anyway. Thanks for catching that again here.

**Pull request merged**